src/Matcher/Pattern/Expander/Repeat.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Coduo\PHPMatcher\Matcher\Pattern\Expander;
+
+use Coduo\PHPMatcher\Factory\SimpleFactory;
+use Coduo\PHPMatcher\Matcher;
+use Coduo\PHPMatcher\Matcher\Pattern\PatternExpander;
+use Coduo\ToString\StringConverter;
+
+final class Repeat implements PatternExpander
+{
+    const NAME = 'repeat';
+
+    /**
+     * @var null|string
+     */
+    private $error;
+
+    /**
+     * @var string
+     */
+    private $pattern;
+
+    /**
+     * @var bool
+     */
+    private $isStrict;
+
+    /**
+     * @var bool
+     */
+    private $isScalar;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function is($name)
+    {
+        return self::NAME === $name;
+    }
+
+    /**
+     * @param $value
+     */
+    public function __construct($pattern, $isStrict = true)
+    {
+        if (!is_string($pattern)) {
+            throw new \InvalidArgumentException("Repeat pattern must be a string.");
+        }
+
+        $this->pattern = $pattern;
+        $this->isStrict = $isStrict;
+        $this->isScalar = true;
+
+        $json = json_decode($pattern, true);
+
+        if ($json !== null && json_last_error() === JSON_ERROR_NONE) {
+            $this->pattern = $json;
+            $this->isScalar = false;
+        }
+    }
+
+    /**
+     * @param $values
+     * @return bool
+     */
+    public function match($values)
+    {
+        if (!is_array($values)) {
+            $this->error = sprintf("Repeat expander require \"array\", got \"%s\".", new StringConverter($values));
+            return false;
+        }
+
+        $factory = new SimpleFactory();
+        $matcher = $factory->createMatcher();
+
+        if ($this->isScalar) {
+            return $this->matchScalar($values, $matcher);
+        }
+
+        return $this->matchJson($values, $matcher);
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getError()
+    {
+        return $this->error;
+    }
+
+    /**
+     * @param array $values
+     * @param Matcher $matcher
+     * @return bool
+     */
+    private function matchScalar(array $values, Matcher $matcher)
+    {
+        foreach ($values as $index => $value) {
+            $match = $matcher->match($value, $this->pattern);
```

Comment:
If we only match value, is following pattern failing ?

`@array@.repeat({"foo": "@string@"})`

```json
[
  {
    "foo": "bar" 
  },
  {
    "test": "bar"
  }
]
``

Reply:
Ok this is covered by https://github.com/coduo/php-matcher/pull/109/files#diff-a3f73e5a606230d5e7eefa13dddd6221R132

Reply:
I'm not sure to understand what you're talking about here ?

Reply:
Just ignore this comment